src/ride/track.c

```diff
@@ -4880,15 +4881,310 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+    // or      ax, cx
+    // mov     ss:word_1362948[ebp], ax
+    RCT2_GLOBAL(0x1362948, uint16) = (uint16) (((eax & 0xFFFF) >> 5) & ((ecx << 3) & 0xFFFF));
+
+    // pop     ecx
+    // pop     eax
+
+    return loc_6CDB35();
+}
+
+money32 do_game_command_set_maze_track(uint16 x, uint8 flags, uint8 direction, uint16 y, uint8 rideIndex, uint8 mode, uint16 unknown, uint16 z) {
+
+	// 0x006CD8CE: game_command_set_maze_track
+	RCT2_GLOBAL(RCT2_ADDRESS_NEXT_EXPENDITURE_TYPE, uint8) = 0;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMAND_MAP_X, uint16) = x + 8;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMAND_MAP_Y, uint16) = y + 8;
+
+	RCT2_GLOBAL(0xF4413E, money32) = 0;
+
+	if (!sub_68B044()) {
+		// loc_6CDE51:
+		return MONEY32_UNDEFINED;
+	}
+
+	if (!(flags & GAME_COMMAND_FLAG_ALLOW_DURING_PAUSED) && RCT2_GLOBAL(RCT2_ADDRESS_GAME_PAUSED, uint8) != 0) {
+		// loc_6CDE2D:
+		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, unsigned short) = STR_CONSTRUCTION_NOT_POSSIBLE_WHILE_GAME_IS_PAUSED;
+		// loc_6CDE51:
+		return MONEY32_UNDEFINED;
+	}
+
+	// loc_6CD920:
+	if ((z & 0xF) != 0) {
+		// loc_6CDE38:
+		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, unsigned short) = 954;
+		// loc_6CDE51:
+		return MONEY32_UNDEFINED;
+	}
+
+	if ((flags & GAME_COMMAND_FLAG_APPLY) != 0) {
+		if ((flags & GAME_COMMAND_FLAG_ALLOW_DURING_PAUSED) == 0) {
+			footpath_remove_litter(x, y, z);
+			map_remove_walls_at((uint16) floor2(x, 32), (uint32) floor2(y, 32), z, z + 0x20);
+		}
+	}
+
+	// loc_6CD954:
+	if (map_is_location_owned(floor2(x, 32), floor2(y, 32), z) == 0) {
+		// loc_6CDE51
+		return MONEY32_UNDEFINED;
+	}
+
+	rct_map_element *mapElement = map_get_first_element_at(x / 32, y / 32);
+
+	if (map_element_get_type(mapElement) != 0) {
+		while (!map_element_is_last_for_tile(mapElement++)) {
+			if ((map_element_get_type(mapElement) == 0)) {
+				break;
+			}
+		}
+	}
+
+	// loc_6CD99A:
+	uint16 temp_dx = z;
+	temp_dx += 0x20;
+	temp_dx = temp_dx >> 3;
+	uint8 temp_dl = (uint8) (temp_dx & 0xFF);
+
+	temp_dl -= mapElement->base_height;
+	if (temp_dl >= 0) {
+		temp_dl = temp_dl >> 1;
+
+		if (temp_dl > RCT2_GLOBAL(0x97D21A + 0x9E, uint8)) {
+			RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = STR_TOO_HIGH_FOR_SUPPORTS;
+			// loc_6CDE51
+			return MONEY32_UNDEFINED;
+		}
+	}
+
+	// loc_6CD9C5
+	// push    eax
+	// push    ebx
+	// push    ecx
+	// push    edx
+	// push    edi
+
+	// mov     bl, dl
+	uint8 bl = rideIndex;
+	// mov     dx, di
+	uint16 temp_dx = z;
+	// shr     dx, 3
+	temp_dx = temp_dx >> 3;
+
+	rct_map_element *mapElement = map_get_first_element_at(x / 32, y / 32);
+
+	do {
+		// loc_6CD9F0:
+		if (map_element_get_type(mapElement) == 8) {
+			if ((temp_dx & 0xff) == mapElement->base_height) {
+				// cmp     bl, [esi+7]
+				// jnz     short loc_6CDA0C
+				if (rideIndex == mapElement->properties.track.ride_index) {
+					// cmp     byte ptr [esi+4], 65h ; 'e'
+					// jz      loc_6CDB35
+					if (mapElement->properties.track.type == 0x65) {
```

Comment:
0x65 is a known track elem type but in this case mazes dont conform to the normal track elem enum names so might be an idea to add another entry to the enum for 0x65 (TRACK_ELEM_INVERTED_90_DEG_UP_TO_FLAT_QUARTER_LOOP).